code/modules/mob/living/carbon/carbon.dm

```diff
@@ -251,17 +251,22 @@
 					"<span class='notice'>You pat [src] on the head.</span>",\
 					)
 				else
-
-					M.visible_message(\
-					"<span class='notice'>[M] gives [src] a [pick("hug","warm embrace")].</span>",\
-					"<span class='notice'>You hug [src].</span>",\
-					)
-					if(ishuman(src))
-						var/mob/living/carbon/human/H = src
-						if(H.wear_suit)
-							H.wear_suit.add_fingerprint(M)
-						else if(H.w_uniform)
-							H.w_uniform.add_fingerprint(M)
+					if(M.zone_selected == ("r_arm" || "r_hand" || "l_arm" || "l_hand"))
```

Comment:
This wont work as expected. It will be the same as `M.zone_selected == "r_arm"`. A switch can do this but here you need an is in list check or if else chain

Reply:
This'll work for Farie's suggestion:
```suggestion
					if(M.zone_selected in list("r_arm", "r_hand", "l_arm", "l_hand"))
```